lib/floe/workflow/choice_rule/data.rb

```diff
@@ -4,88 +4,36 @@ module Floe
   class Workflow
     class ChoiceRule
       class Data < Floe::Workflow::ChoiceRule
-        def true?(context, input)
-          lhs = variable_value(context, input)
-          rhs = compare_value(context, input)
-
-          validate!(lhs)
-
-          case compare_key
-          when "IsNull" then is_null?(lhs)
-          when "IsPresent" then is_present?(lhs)
-          when "IsNumeric" then is_numeric?(lhs)
-          when "IsString" then is_string?(lhs)
-          when "IsBoolean" then is_boolean?(lhs)
-          when "IsTimestamp" then is_timestamp?(lhs)
-          when "StringEquals", "StringEqualsPath",
-               "NumericEquals", "NumericEqualsPath",
-               "BooleanEquals", "BooleanEqualsPath",
-               "TimestampEquals", "TimestampEqualsPath"
-            lhs == rhs
-          when "StringLessThan", "StringLessThanPath",
-               "NumericLessThan", "NumericLessThanPath",
-               "TimestampLessThan", "TimestampLessThanPath"
-            lhs < rhs
-          when "StringGreaterThan", "StringGreaterThanPath",
-               "NumericGreaterThan", "NumericGreaterThanPath",
-               "TimestampGreaterThan", "TimestampGreaterThanPath"
-            lhs > rhs
-          when "StringLessThanEquals", "StringLessThanEqualsPath",
-               "NumericLessThanEquals", "NumericLessThanEqualsPath",
-               "TimestampLessThanEquals", "TimestampLessThanEqualsPath"
-            lhs <= rhs
-          when "StringGreaterThanEquals", "StringGreaterThanEqualsPath",
-               "NumericGreaterThanEquals", "NumericGreaterThanEqualsPath",
-               "TimestampGreaterThanEquals", "TimestampGreaterThanEqualsPath"
-            lhs >= rhs
-          when "StringMatches"
-            lhs.match?(Regexp.escape(rhs).gsub('\*', '.*?'))
-          else
-            raise Floe::InvalidWorkflowError, "Invalid choice [#{compare_key}]"
+        DATA_RULES = {
+          "IsNull"            => Floe::Workflow::ChoiceRule::IsNull,
+          "IsPresent"         => Floe::Workflow::ChoiceRule::IsPresent,
+          "IsNumeric"         => Floe::Workflow::ChoiceRule::IsNumeric,
+          "IsString"          => Floe::Workflow::ChoiceRule::IsString,
+          "IsBoolean"         => Floe::Workflow::ChoiceRule::IsBoolean,
+          "IsTimestamp"       => Floe::Workflow::ChoiceRule::IsTimestamp,
+          "Equals"            => Floe::Workflow::ChoiceRule::Data::Equals,
+          "LessThan"          => Floe::Workflow::ChoiceRule::Data::LessThan,
+          "GreaterThan"       => Floe::Workflow::ChoiceRule::Data::GreaterThan,
+          "LessThanEquals"    => Floe::Workflow::ChoiceRule::Data::LessThanEquals,
+          "GreaterThanEquals" => Floe::Workflow::ChoiceRule::Data::GreaterThanEquals,
+          "Matches"           => Floe::Workflow::ChoiceRule::Data::Matches
```

Comment:
Is this right?  I don't see a Data namespace.

Reply:
what the heck?
why are these tests running?

Reply:
looks like it was finding `Floe::Workflow::ChoiceRule::Matches` when referencing `Floe::Workflow::ChoiceRule::Data::Matches`